### DISCUSSION TITLE
Is it real to get unique(or 80% unique) values for for faker provider in output?

### DISCUSSION THREAD

**Discussion by alex-tomson** [rheinwerk-verlag/pganonymize#42] [General]
I'm getting the same name for entire column when use faker.first_name. Is it real to get different values to each row?
Maybe I'm doing something wrong, so give me advise please, how I can get different values for faker provider

**Comment by alex-tomson** [comment]
After some time diving in codebase I find quick solution
In file pganonymize/utils.py change line

> data = parmap.map(process_row, records, columns, excludes, pm_pbar=verbose)

to

> data = parmap.map(process_row, records, columns, excludes, pm_pbar=verbose, pm_parallel=False)

Seems like force parallelization going bad for faker provider

**Reply by hkage** [reply]
Hi, sorry I didn't have the time to take a look into your issue. I must admit that most of the recent changes, including the usage of `parmap`, are features I haven't tested so much, because we still don't use `pganonymize` in production and I can only develop and improve this project in my spare time. 

Afaik `pm_parallel=False` will deactivate the parallelization, which completely negotiates the usage of `parmap`. Therefore I cannot simply disable it for now. I think I will have to take a deeper look into `faker` in combination with parallelization, otherwise I will have to add an option to disable it.

EDIT: Issue https://github.com/rheinwerk-verlag/pganonymize/issues/31 describes the same behaviour, so it seems to be a general issue with `faker` and `parmap`.
